homeassistant/components/rfxtrx/__init__.py

```diff
@@ -405,6 +412,31 @@ def get_device_id(device, data_bits=None):
     return (f"{device.packettype:x}", f"{device.subtype:x}", id_string)
 
 
+async def force_entity_name(
+    hass: HomeAssistantType, domain: str, unique_id: str, name: str
+):
+    """Force the user name to name set in config.
+
+    This is a temporary measure to ensure old configurations get upgraded to
+    new format.
+    """
+
+    reg = await async_get_entity_registry(hass)
+    entry = reg.async_get_or_create(domain, DOMAIN, unique_id, suggested_object_id=name)
+    entity_id = f"{domain}.{slugify(name)}"
+
+    updates = {}
+    if entry.name != name:
```

Comment:
We should only change the `entry.name` if it's `None`. If it's something else it's been set by the user.

Reply:
Once the option flow is in that is valid. Currently the config from yaml is master that is why I did it this way. But sure they can can remove the name if they want the config from yaml to take effect.

Reply:
The user can already change the name in the entity registry regardless of config yaml. If that has been done, we should honor that.

Reply:
The point is if they have *never" set it via GUI and then change the name in yaml. If I change this their yaml change won't take effect and they have no idea why.

Reply:
The `entry.name` attribute will not be set unless the user has changed it in the entity registry, if I'm reading the entity registry code correctly. The default will be `None` for entry.name` if the user hasn't changed it via GUI. The `original_name` attribute will hold the original name that is passed from the enity `name` attribute.

Am I missing something?

Reply:
This pull request set that exact same GUI value based on the yaml config. That way when yaml later is removed, the config value will remain. If we set original value, when yaml is removed all entities would revert to a default name.

We want to import yaml set name as a pree-seed of the GUI value.

Reply:
Yes, but only if the user hasn't already changed name in the entity registry. That's why we should check for `None` value here. If the user has changed name in the entity registry, that name will always be restored regardless of config yaml is removed.

Reply:
I think missunderstood you. I thought you talked about `entry.name`. `name` can't be None here. Function is not called if the name is none/missing in yaml.

Reply:
I'm talking about `entry.name` ie the name attribute of the entity registry entry.

Reply:
Then my comment above stand. It becomes a one time import. And yaml name ends up ignored. Imho we should respect changes in yaml untill we are done with conversion to config flow.

Reply:
When we respect yaml and the user changes the name in the UI, it won't be honored after restart, so one way or another, changing it somewhere will lead to confusing behavior. I would say it makes more sense then to respect changes made from the UI and document that the Name parameter is a one-time-thing.

Reply:
It is possible to remove it from yaml though. But okey if you both want it that way. For me it's more confusing to ignore something explicit in yaml.. But maybe I'm yaml old fashioned. :)

Reply:
Yes, YAML will be a one time import, like we have already discussed.

Reply:
This PR won't go out in a patch release, so we should have time do the config flow with user step etc.

Reply:
It's not now thou, yaml is always imported.. it won't be onetime until the option flow change merges. But I digress, i'm tired of the discussion so i'll yield.

Reply:
Wait.. What about entity_id then? I can't ignore yaml name and then update the entity_id based on that name.. I'll just give up trying to make peoples migration easier.

Reply:
To make the migration as transparent as possible for the user, I think we should instead ask the user in the config flow if they want to migrate. If they say yes, we overwrite both name and entity_id with the yaml as base. If they say no, we don't do any migration at all.

We can add a migrate step that is entered from the import step, and the user will need to complete that step before finishing the import flow.